### ISSUE TITLE
Conditions on two hyperparameters: Encoding Intervals Hyperparams

### ISSUE THREAD

**Issue by gamer01** [automl/ConfigSpace#131]
# Problem:

i want to generate ranges a,b with in the boundaries A and B, where A<=a, a<=b and b<=B.

I tried to model that with the help of conditions, but i found no way, as conditions can only depend on one parent hyperparameter, but not on the relation of two hyperparameters.

# Workaround:
in my case all ranges are constants, therefore i can do the following
```
CategoricalHyperparameter('ab', [(a,b) for a in range(A,B+1) for b in range(A,B+1) if a<=b])
```

However this can create a huge amount of potential values and is generally not posible for real valued intervals.

# Feature Request:
let a,b,c be hyperparameters
```
conditions.LessThanCondition(c,a,b)
```

currently b needs to be a constant.

**Comment by mfeurer** [comment]
I think the easiest way to add a feature like this would be to implement a `LargerThanForbiddenClause` which rejects configurations for which the value of hyperparameter a is larger than the value of hyperparameter b. Please see [here](https://automl.github.io/ConfigSpace/master/API-Doc.html#forbidden-clauses) for more information on forbidden clauses.

PR welcome.

**Comment by gamer01** [comment]
The documentation indicates that the ForbiddenClause can only depend on constants, what i need is a dynamic version, that will be able to handle relations of hyperparameters.

**Comment by mfeurer** [comment]
Yes, currently, this is not possible. The `LargerThanForbiddenClause` does not exist yet.
